client/src/components/categorical/category.js

```diff
@@ -238,58 +238,66 @@ class Category extends React.Component {
               {""}
             </label>
             {/* Dialog uses portal, can be anywhere/factored out */}
-            <Dialog
-              icon="tag"
-              title="Edit category name"
-              isOpen={
-                annotations.isEditingCategoryName &&
-                annotations.categoryEditable === metadataField
-              }
-              onClose={this.disableEditCategoryMode}
-            >
-              <div className={Classes.DIALOG_BODY}>
-                <div style={{ marginBottom: 20 }}>
-                  <p>New, unique category name:</p>
-                  <InputGroup
-                    autoFocus
-                    onChange={e =>
-                      this.setState({ newCategoryText: e.target.value })
-                    }
-                    leftIcon="tag"
-                  />
-                </div>
-              </div>
-              <div className={Classes.DIALOG_FOOTER}>
-                <div className={Classes.DIALOG_FOOTER_ACTIONS}>
-                  <Tooltip content="Close this dialog without editing the category.">
-                    <Button onClick={this.disableEditCategoryMode}>
-                      Cancel
-                    </Button>
-                  </Tooltip>
-                  <Button
-                    disabled={newCategoryText.length === 0}
-                    onClick={this.handleEditCategory}
-                    intent="primary"
-                  >
-                    Edit category name
-                  </Button>
-                </div>
-              </div>
-            </Dialog>
+
             <span
               data-testid={`category-expand-${metadataField}`}
               style={{
                 cursor: "pointer",
                 display: "inline-block"
               }}
               onClick={() => {
-                this.setState({ isExpanded: !isExpanded });
+                const editingCategory =
+                  annotations.isEditingCategoryName &&
+                  annotations.categoryEditable === metadataField;
+                if (!editingCategory) {
+                  this.setState({ isExpanded: !isExpanded });
+                }
               }}
             >
               {isUserAnno ? (
                 <Icon style={{ marginRight: 5 }} icon="tag" iconSize={16} />
               ) : null}
-              {metadataField}
+
+              {annotations.isEditingCategoryName &&
+              annotations.categoryEditable === metadataField ? (
```

Comment:
the name `categoryEditable` is confusing -- if I read the code right, it contains the name of the category currently being edited, ie, the category in "editable mode".    Consider renaming to `.categoryBeingEdited` or some such.

Reply:
Will change